app/models/user.rb

```diff
@@ -41,7 +43,43 @@ def self.find_first_by_auth_conditions(warden_conditions)
     end
   end
 
+  def active?
+    !deactivated_at
+  end
+
+  def deactivate!
```

Comment:
Do you have specs that show that these update the Agents' `deactivated` columns?